roborazzi/src/main/java/com/github/takahirom/roborazzi/Roborazzi.kt

```diff
@@ -429,6 +432,24 @@ fun ViewInteraction.captureAndroidView(
   )
 }
 
+private fun MutableList<RoboCanvas>.addIfChanged(
+  next: RoboCanvas,
+  roborazziOptions: RoborazziOptions
+) {
+  val prev = this.lastOrNull() ?: run {
+    this.add(next)
+    return
+  }
+  val differ: ImageComparator.ComparisonResult =
+    prev.differ(next, 1.0)
+  if (!roborazziOptions.compareOptions.resultValidator(differ)) {
+    this.add(next)
```

Comment:
Should this be configurable? I'd like to be able to use this for capturing frame by frame animations.

Reply:
You have a point, but I think this implementation is better than the current one. I'll create an issue for this. 👍